### PR TITLE
Remove credential subject

### PR DESCRIPTION
**Pull request by vineetpant**
## Descripton

credential_subject.id  has been removed from vade-evan-bbs and vade-jwt-vc , the functions and payloads need to be updated to reflect the change.

## Details

- Add subject_id param to helper functions where subject_did is required
- Adjust payload to functions where subject_did is passed
- Adjust test cases
- Update c-lib, wasm, cli bindings

### CODE REVIEW

File: src/api/vade_evan_api.rs
Comment (Tschuck) [inline src/api/vade_evan_api.rs]:
```suggestion
    ///         const PROVER_DID: &str = "did:evan:EiAee4ixDnSP0eWyp0YFV7Wt9yrZ3w841FNuv9NSLFSCVA";
```

File: src/api/vade_evan_api.rs
Comment (Tschuck) [inline src/api/vade_evan_api.rs]:
```suggestion
    ///                   PROVER_DID,
```

File: src/helpers/credential.rs
Comment (Tschuck) [inline src/helpers/credential.rs:1019]: Id is still there in line 1065.

`"id": "did:evan:EiAOD3RUcQrRXNZIR8BIEXuGvixcUj667_5fdeX-Sp3PpA",`

File: src/helpers/presentation.rs
Comment (Tschuck) [inline src/helpers/presentation.rs]:
```suggestion
```

**Comment by vineetpant** [comment]
aligned vade_evan for new vade_bbs structures, @S3bb1  @wulfraem @Tschuck

File: Cargo.toml
Comment (wulfraem) [inline Cargo.toml]: Just as a reminder before merging this MR too early: This should be updated after merging the [related MR](https://github.com/evannetwork/vade-jwt-vc/pull/7).

File: Cargo.toml
Comment (wulfraem) [inline Cargo.toml]: Just as a reminder before merging this MR too early: This should be updated after merging the [related MR](https://github.com/evannetwork/vade-evan-bbs/pull/31).

File: src/helpers/credential.rs
Comment (wulfraem) [inline src/helpers/credential.rs]: Do we still need it? ^^

File: src/helpers/presentation.rs
Comment (wulfraem) [inline src/helpers/presentation.rs]:
```suggestion
    }"###;
```

^^

**Review by wulfraem — CHANGES_REQUESTED** [review]

File: Cargo.toml
Comment (vineetpant) [reply Cargo.toml]: yes, i am waiting for `credential_subject_id remove` branch to be merged so that it can be updated.

File: Cargo.toml
Comment (vineetpant) [reply Cargo.toml]: yes, i am waiting for credential_subject_id remove branch to be merged so that it can be updated.

File: src/helpers/credential.rs
Comment (vineetpant) [reply src/helpers/credential.rs]: removed the unused commented code

**Review by wulfraem — APPROVED** [review]

**Review by S3bb1 — APPROVED** [review]

**Pull request merged**
